spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/core/CredentialsProviderAutoConfiguration.java

```diff
@@ -101,4 +112,18 @@ private static ProfileCredentialsProvider createProfileCredentialProvider(Profil
 		return ProfileCredentialsProvider.builder().profileName(profile.getName()).profileFile(profileFile).build();
 	}
 
+	private static boolean shouldCreateStsIdentityTokenCredentialsProvider() {
+		return ClassUtils.isPresent(STS_WEB_IDENTITY_TOKEN_FILE_CREDENTIALS_PROVIDER, null)
+			&& StringUtils.hasText(System.getenv(AWS_ROLE_ARN_ENV_VARIABLE))
```

Comment:
Do these properties make sense only as env variables or we should also include them in `CredentialsProperties`?